server/BusinessLogic/ProductManagement/DTOs/CreateProductDTO.cs

```diff
@@ -0,0 +1,13 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace CoffeeShop.BusinessLogic.ProductManagement.DTOs;
+
+public sealed record CreateProductDTO(
+    string Name,
+    string Description,
+    [RegularExpression(@"^\d+(\.\d{1,2})?$", ErrorMessage = "Price can have up to two decimal places.")]
```

Comment:
This validation weird - we are using decimals, therefore we will always more digits due to rounding errors e.g. 2.000001. I suggest just rounding it to cents in business logic